grails-app/services/org/broadinstitute/orsp/StorageProviderService.groovy

```diff
@@ -238,8 +239,8 @@ class StorageProviderService implements Status {
         if (!document.mimeType) {
             throw new IllegalArgumentException("Mime Type is required")
         }
-        if (!document.projectKey) {
-            throw new IllegalArgumentException("Project Key is required")
+        if (!document.projectKey && !document.consentCollectionLinkId) {
+            throw new IllegalArgumentException("Project Key or consentCollectionLink must be specified")
```

Comment:
Why are we enforcing this here? What value does this provide to the application to actively prevent these two values from being set at the same time? Let me add here that there is no **business requirement** that this restriction be in place.

Reply:
ok I can remove this validation, the idea was to validate that project key or consent key are not empty, if project key and consent collection link are null at the same time, doesn't make any sence to save a document without any reference.

Reply:
Pretty sure that I misread this code before - I thought you were checking to ensure that **only** one or the other was populated, but not both of them populated together. This is what I meant as not being a "business requirement". 

What you describe **does** make sense, there should be at least one OR the other populated, both is OK, but both empty is not OK. So, please accept my apologies ... we should have this error condition in place.

Reply:
Thank you for adding this back in 😄